src/mokka/src/integration-test/java/pl/hycom/mokka/emulator/mock/MockConfigurationControllerIT.java

```diff
@@ -0,0 +1,179 @@
+public class MockConfigurationControllerIT {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private MockConfigurationRepository mockConfigurationRepository;
+
+    @Autowired
+    private WireMockServer wm;
+
+    private static final String name = "test";
+    private static final String httpMethod = "GET";
+    private static final int status = 303;
+    private static final String path = "path";
+    private static final String slash = "/";
+
+    private String id;
+    private String uuid;
+
+    @BeforeEach
+    public void setUp() throws Exception{
+        RestAssured.port = wm.port();
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void create_mock() throws Exception {
+
+        addMock();
+
+        Response response = RestAssured.get("/__admin/mappings/" + uuid);
+        expect().statusCode(200).contentType(ContentType.JSON).response();
+
+        Assertions.assertEquals(response.jsonPath().get("name"), name);
+        Assertions.assertEquals(response.jsonPath().get("request.url"), slash + path);
+        Assertions.assertEquals(response.jsonPath().get("request.method"), httpMethod);
+
+        mvc.perform(asyncRequest(delete("/configuration/" + id)).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void update_mock() throws Exception {
+
+        addMock();
+        String testChange = "test_change";
+
+        String requestBody = "{\"status\":" + status + ",\"httpMethod\":\"" + httpMethod + "\",\"path\":\""+path+"\",\"id\":"+id+",\"description\":\"test\",\"pattern\":null,\"name\":\"" + testChange + "\",\"enabled\":true,\"proxyBaseUrl\":null,\"timeout\":0,\"order\":0,\"configurationContent\":{\"string\":{\"value\":\"d\"}},\"updated\":\"2020-03-18T01:44:39.089+0000\",\"errors\":null,\"changes\":null,\"showChanges\":false,\"showConfiguration\":true,\"type\":\"string\"}";
+
+        mvc.perform(asyncRequest(put("/configuration")).contentType(MediaType.APPLICATION_JSON).with(csrf())
+            .content(requestBody))
+            .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings/" + uuid);
+        expect().statusCode(200).contentType(ContentType.JSON).response();
+
+        Assertions.assertEquals(response.jsonPath().get("name"), testChange);
+        Assertions.assertEquals(response.jsonPath().get("request.url"), slash + path);
+        Assertions.assertEquals(response.jsonPath().get("request.method"), httpMethod);
+
+        mvc.perform(asyncRequest(delete("/configuration/" + id)).contentType(MediaType.APPLICATION_JSON).with(csrf()))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @WithMockUser(roles="ADMIN")
+    public void disable_mock() throws Exception {
+
+        addMock();
+
+        mvc.perform(asyncRequest(post("/configuration/" + id +"/disable")).contentType(MediaType.APPLICATION_JSON).with(csrf()))
```

Comment:
`post("/configuration/{id}/disable", id)` - should work also and is more readable